server/controllers/notification.js

```diff
@@ -0,0 +1,50 @@
+const { Notification } = require("../models/Notification");
+const asyncHandler = require("express-async-handler");
+
+exports.newNotification = asyncHandler(async (req, res) => {
+  const { type, title, description, link, user } = req.body;
+
+  const newNotification = await Notification.create({
+    type: type,
+    title: title,
+    description: description,
+    link: link,
+    user: user,
+  });
+
+  if (newNotification) return res.status(200).json({ data: newNotification });
+
+  res.status(400);
+  throw new Error("Notification could not be created!");
+});
+
+exports.getAllNotifications = asyncHandler(async (req, res) => {
+  const notifications = await Notification.find({ user: req.user.id });
+
+  return res.status(200).json({ data: notifications });
+});
+
+exports.getUnreadNotifications = asyncHandler(async (req, res) => {
+  const notifications = await Notification.find({
+    read: false,
+    user: req.user.id,
+  });
+
+  return res.status(200).json({ data: notifications });
+});
+
+exports.setNotificationAsRead = asyncHandler(async (req, res) => {
+  const { id } = req.body;
+
+  const notification = await Notification.findOneAndUpdate(
+    { _id: id },
+    {
+      $set: { read: true },
+    },
+    { new: true }
+  );
```

Comment:
I think this looks good for methods in the controller for now. Right now, I'm doing some of this logic (filter by currentUser/read) in the front end as well since I'm just reading from a dummy data file I made. When we integrate the fe/be for this feature, I like the idea of doing this all on the back end.

Reply:
Sure, that sounds like a good idea.